omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,359 @@
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	/**
+	 *  Delete Allergy incorrectly
+	 *  throw ObjectNotFoundException because allergy does not exist
+	 */
+	@Test(expected = ObjectNotFoundException.class)
+	public void shouldThrowExceptionWhenDeletingNonExistentAllergy() throws Exception
+	{
+		// attempt to delete allergy
+		handle(newDeleteRequest(getURI() + "/" + "NonExistentAllergyUUID", new Parameter("reason", "unit test")));
+	}
+
+	/**
+	 *  Delete all Allergies
+	 */
+	@Test
+	public void shouldDeleteAllAllergies() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		Allergies allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+		Assert.assertFalse(allergy.isVoided());
+
+		// attempt to delete all allergies
+		handle(newDeleteRequest(getURI(), new Parameter("reason", "unit test")));
+		
+		allergy = Context.getPatientService().getAllergyByUuid(getUuid());		
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+	}
+
+	/**
+	 *  Test Allergy for correct values 
+	 *  - does not compare reactions
+	 */
+	public void testAllergyForEquality(Allergy allergy, SimpleObject so)
+	{
+		Assert.assertEquals(allergy.getComment(), Util.getByPath(so, "comment"));
+		Assert.assertEquals(allergy.getSeverity().getUuid(), Util.getByPath(so, "severity/uuid"));
+
+		if (allergy.getAllergen().isCoded()) {
+			Assert.assertEquals(allergy.getAllergen().getCodedAllergen().getUuid(), Util.getByPath(so, "allergen/codedAllergen/uuid"));
+		} else {
+			Assert.assertEquals(allergy.getAllergen().getCodedAllergen().getUuid(), Util.getByPath(so, "allergen/codedAllergen/uuid"));
+			Assert.assertEquals(allergy.getAllergen(), Util.getByPath(so, "allergen/nonCodedAllergen"));
+		}
+		Assert.assertEquals(allergy.getAllergen().getAllergenType().toString(), Util.getByPath(so, "allergen/allergenType"));
+	}
+
+	/**
+	 *  Test Allergy for correct values 
+	 *  - does not compare allergyReactionId
+	 */
+	public void testAllergyForEquality(SimpleObject so, boolean isCodedAllergen)
+	{
+		Assert.assertEquals("allergy comment", Util.getByPath(so, "comment"));
+		Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(so, "severity/uuid"));
+
+		if (isCodedAllergen) {
+			Assert.assertEquals("35d3346a-6769-4d52-823f-b4b234bac3e3", Util.getByPath(so, "allergen/codedAllergen/uuid"));
+		} else {
+			Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(so, "allergen/codedAllergen/uuid"));

```

Comment:
this just puts the assertions in a separate method so the test method is smaller